Generals/Code/GameEngine/Source/GameClient/InGameUI.cpp

```diff
@@ -2012,16 +2022,20 @@ void InGameUI::messageColor( const RGBColor *rgbColor, UnicodeString format, ...
 
 	// construct the final text after formatting
 	va_list args;
-  va_start( args, format );
+	va_start( args, format );
 	WideChar buf[ UnicodeString::MAX_FORMAT_BUF_LEN ];
-  if( _vsnwprintf(buf, sizeof( buf )/sizeof( WideChar ) - 1, format.str(), args ) < 0 )
-			throw ERROR_OUT_OF_MEMORY;
-	formattedMessage.set( buf );
-  va_end(args);
-
-	// add the text to the ui
-	addMessageText( formattedMessage, rgbColor );
+	int result = _vsnwprintf(buf, sizeof( buf )/sizeof( WideChar ) - 1, format.str(), args );
```

Comment:
A mistake creeped in here. va_end(args) was lost by mistake. I will fix in follow up.